components/BottomNavigation/src/MDCBottomNavigationBar.m

```diff
@@ -0,0 +1,354 @@
+
+@property(nonatomic, assign) BOOL itemsDistributed;
+@property(nonatomic, assign) BOOL titleBelowItem;
+@property(nonatomic, strong) NSMutableArray<MDCBottomNavigationItemView *> *itemViews;
+@property(nonatomic, strong) UIView *containerView;
+
+@end
+
+@implementation MDCBottomNavigationBar
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCBottomNavigationBarInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCBottomNavigationBarInit];
+  }
+  return self;
+}
+
+- (void)commonMDCBottomNavigationBarInit {
+  self.backgroundColor = [UIColor whiteColor];
+  self.autoresizingMask = (UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleWidth);
+  _selectedItemTintColor = [UIColor blackColor];
+  _unselectedItemTintColor = [UIColor grayColor];
+  _titleHideState = MDCBottomNavigationBarTitleHideStateDefault;
+  _landscapeItemMode = MDCBottomNavigationBarLandscapeItemModeDistributeCenteredTitles;
+  _itemsDistributed = YES;
+  _titleBelowItem = YES;
+  _maxLandscapeContainerWidth = kMDCBottomNavigationBarLandscapeContainerWidth;
+  _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+  _containerView.autoresizingMask = (UIViewAutoresizingFlexibleLeftMargin |
+                                     UIViewAutoresizingFlexibleRightMargin);
+  [self addSubview:_containerView];
+  _itemViews = [NSMutableArray array];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  CGSize size = self.bounds.size;
+  if (UIDeviceOrientationIsLandscape([UIDevice currentDevice].orientation)) {
+    [self layoutLandscapeModeWithSize:size width:self.maxLandscapeContainerWidth];
+  } else {
+    [self sizeContainerViewItemsDistributed:YES withSize:size width:size.width];
+    [self showTitleBelowItem:YES];
+  }
+  [self layoutItemViewsWithLayoutDirection:self.mdc_effectiveUserInterfaceLayoutDirection];
+}
+
+- (CGSize)sizeThatFits:(CGSize)size {
+  UIEdgeInsets insets = [self getUIEdgeInsets];
+  CGFloat heightWithInset = kMDCBottomNavigationBarHeight + insets.bottom;
+  return CGSizeMake(size.width, heightWithInset);
+}
+
+- (void)layoutLandscapeModeWithSize:(CGSize)size width:(CGFloat)width {
+  switch (self.landscapeItemMode) {
+    case MDCBottomNavigationBarLandscapeItemModeDistributeCenteredTitles:
+      [self sizeContainerViewItemsDistributed:YES withSize:size width:width];
+      [self showTitleBelowItem:YES];
+      break;
+    case MDCBottomNavigationBarLandscapeItemModeDistributeAdjacentTitles:
+      [self sizeContainerViewItemsDistributed:YES withSize:size width:width];
+      [self showTitleBelowItem:NO];
+      break;
+    case MDCBottomNavigationBarLandscapeItemModeCluster:
+      [self sizeContainerViewItemsDistributed:NO withSize:size width:width];
+      [self showTitleBelowItem:YES];
+      break;
+  }
+}
+
+- (UIEdgeInsets)getUIEdgeInsets {
```

Comment:
super-nit: This is not an Objective-c~ish method name.  Objective-C rarely uses get.  
This is also a invoking safeAreaInsets.
Consider declaring as a property :
@property (readonly) UIEdgeInsets mdc_safeAreaInsets;

Naming is hard!